x-pack/legacy/plugins/apm/public/components/shared/charts/CustomPlot/StaticPlot.js

```diff
@@ -138,21 +139,53 @@ class StaticPlot extends PureComponent {
     }
   }
 
+  /**
+   * A tick format function that takes the timezone from Kibana's settings into
+   * account. Used if no tickFormatX prop is supplied.
+   *
+   * This produces the same results as the built-in formatter from D3, which is
+   * what react-vis uses, but shifts the timezone.
+   */
+  tickFormatX = value => {
+    const xDomain = this.props.plotValues.x.domain();
+    const guessedZoneOffset = moment.tz.zone(moment.tz.guess()).parse(value);
+    const configuredZoneOffset = moment.tz
+      .zone(moment.defaultZone.name)
+      .parse(value);
+
+    value.setTime(
+      value.getTime() + (configuredZoneOffset - guessedZoneOffset) * 60000
```

Comment:
Nice idea with just shifting the time, instead of re-implementing the date logic. Is it possible to add a test for this somehow?

Reply:
@dgieselaar since the offsets are calculated based on the time input, those offset values should include/exclude DST as needed and be correct.

Reply:
Oh right, this formats per-value. So that should be fine. We might be missing something though, seems like the native time formatter is a little smarter about how to format. For instance, this is what it looks like with 7 days selected. All labels show '06 AM'.


![image](https://user-images.githubusercontent.com/352732/66936746-80241e00-f03e-11e9-87ce-c5f5be93a8f2.png)

This is what it previously looked like:

![image](https://user-images.githubusercontent.com/352732/66936789-8f0ad080-f03e-11e9-961c-60f201041b2d.png)

Reply:
Hmm thanks for spotting this. I think it might have something to do with the domain/range being provided to the scale. Will look into it.